### PR TITLE
Moments presenter is no longer QObject and View is passed to constructor.

### PR DESCRIPTION
**Pull request by adriazalvarez**
### Description of work
This PR removes the Qt dependency of the presenter of the Moments tab. The implementation follows very closely the layout set in #36478 , and more justification for the changes can be found therein. 
#### Summary of work

Part of  #36477 . 

### To test:
1. Open `Data Manipulation -> Moments` interface. 
2. Load a `sqw` file, for example: 'iris26173_graphite002_sqw.xs'
3. Test that the interface behaves and works as before. Try changing plot limits and property limits, and run the algorithm.
4. Check that plot options are enabled once the plot is finished. 

*This does not require release notes as it is internal maintenance.*

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Review by robertapplin — DISMISSED** [review]
This refactor looks really good, you've aced it! Just one small change but then would be very happy for this to be merged

**Comment by robertapplin** [comment]
Unassigning myself as I am on A/L for rest of the week

File: qt/scientific_interfaces/Inelastic/Manipulation/IMomentsView.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  /// Function to set the range selector on the mini plot
+  virtual void setRangeSelector(const QPair<double, double> &bounds) = 0;
+  /// Sets the min of the range selector if it is less than the max
+  virtual void setRangeSelectorMin(double newValue) = 0;
+  /// Sets the max of the range selector if it is more than the min
+  virtual void setRangeSelectorMax(double newValue) = 0;
+
+  virtual void plotNewData(std::string const &filename) = 0;
+  virtual void replot() = 0;
+  virtual void plotOutput(QString outputWorkspace) = 0;
```
[the remark this answers is not in the document]
Comment (adriazalvarez) [reply qt/scientific_interfaces/Inelastic/Manipulation/IMomentsView.h]: Thanks.

**Review by RichardWaiteSTFC — APPROVED** [review]
Looks good, last commit addressed @robertapplin comment.
I built this, loaded data, moved sliders, scaled, clicked run, save results etc. and everything seemed to work so approving!

**Review by RichardWaiteSTFC — APPROVED** [review]

**Pull request merged**